stdlib/mexpr/eq.mc

```diff
@@ -425,19 +457,32 @@ lang BoolPatEq = BoolPat
     else None ()
 end
 
-lang AndPatEq
+lang AndPatEq = AndPat
   sem eqPat (env : Env) (free : Env) (patEnv : NameEnv) (lhs : Pat) =
-  -- TODO(dlunde,2020-09-29)
+  | PAnd {lpat = l2, rpat = r2} ->
+    match lhs with PAnd {lpat = l1, rpat = r1} then
+      match eqPat env free patEnv l1 l2 with Some (f,p) then
+        eqPat env free patEnv r1 r2
```

Comment:
It seems weird to not use `(f, p)` at all, is that intentional? I'd've thought it would work the same as names in, e.g., `SeqTotPatEq`, where the values are treated as an accumulator, forwarded left to right through the child patterns.

Reply:
I see, then I should left it as it is before (the order of the binding should matter?):
```
match eqPat env free patEnv l1 l2 with Some (free, patEnv) then
        eqPat env free patEnv r1 r2
```
 I made the same mistake in SeqEdgePatEq, I will fix that also. Thanks!